internal/xds/translator/translator.go

```diff
@@ -53,10 +57,41 @@ func (t *Translator) Translate(ir *ir.Xds) (*types.ResourceVersionTable, error)
 		return nil, err
 	}
 
+	// If there is a loaded extension that wants to inject clusters/secrets, then call it
```

Comment:
can we move all extension code into a separate file similar to what's done for ratelimit